src/Components/Common/Buttons/UpButton/UpButton.tsx

```diff
@@ -0,0 +1,35 @@
+import { type FC, type ReactElement, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import classes from './UpButton.module.scss';
+import { EventType } from '../../../../utils/types/enums';
+
+export const UpButton: FC = (): ReactElement => {
+  const upButton = useRef<HTMLButtonElement>(null);
+
+  const onScroll = (): void => {
+    const element: HTMLElement = document.documentElement;
+
+    if (upButton.current) {
+      upButton.current.hidden = element.scrollTop < element.clientHeight;
```

Comment:
Same here, if possible let's move the hidden value to state

Reply:
It is possible. I have done it!